pkg/e2e/harness_runner/harness_runner.go

```diff
@@ -7,34 +7,114 @@ import (
 	"strings"
 
 	"github.com/onsi/ginkgo/v2"
-	"github.com/openshift/osde2e/pkg/common/alert"
+	. "github.com/onsi/gomega"
 	viper "github.com/openshift/osde2e/pkg/common/concurrentviper"
 	"github.com/openshift/osde2e/pkg/common/config"
 	"github.com/openshift/osde2e/pkg/common/helper"
 	"github.com/openshift/osde2e/pkg/common/label"
-	"github.com/openshift/osde2e/pkg/common/prow"
+	"github.com/openshift/osde2e/pkg/common/runner"
+	"github.com/openshift/osde2e/pkg/common/templates"
+	"github.com/openshift/osde2e/pkg/common/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	serviceAccountDir = "/var/run/secrets/kubernetes.io/serviceaccount"
+	TimeoutInSeconds  = viper.GetFloat64(config.Tests.PollingTimeout)
+	harnesses         = strings.Split(viper.GetString(config.Tests.TestHarnesses), ",")
+	h                 *helper.H
+	err               error
+	HarnessEntries    []ginkgo.TableEntry
 )
 
 var _ = ginkgo.Describe("Test Harness", ginkgo.Ordered, label.TestHarness, func() {
-	var h *helper.H
-	ginkgo.BeforeAll(func() {
-		h = helper.New()
-	})
-
-	TimeoutInSeconds := viper.GetFloat64(config.Tests.PollingTimeout)
-	ginkgo.It("should run until completion", func(ctx context.Context) {
-		ginkgo.GinkgoWriter.Write([]byte("Test harness timeout is " + fmt.Sprintf("%v", TimeoutInSeconds)))
-		h.SetServiceAccount(ctx, viper.GetString(config.Tests.TestUser))
-		harnesses := strings.Split(viper.GetString(config.Tests.TestHarnesses), ",")
-		failed := h.RunTests(ctx, "test-harness", int(TimeoutInSeconds), harnesses, []string{})
-		if len(failed) > 0 {
-			message := fmt.Sprintf("Tests failed: %v", failed)
-			if url, ok := prow.JobURL(); ok {
-				message += "\n" + url
-			}
-			if err := alert.SendSlackMessage(viper.GetString(config.Tests.SlackChannel), message); err != nil {
-				log.Printf("Failed sending slack alert for test failure: %v", err)
-			}
-		}
-	}, TimeoutInSeconds+30)
+	for _, harness := range harnesses {
+		HarnessEntries = append(HarnessEntries, ginkgo.Entry("should run "+harness+" successfully", harness))
+	}
+	ginkgo.DescribeTable("Executing Harness",
+		func(harness string) {
```

Comment:
```suggestion
		func(ctx context.Context, harness string) {
```
ginkgo can provide a context to the function that way you don't have to use `context.TODO()` below